CanvasColorSpaceProposal.md

```diff
@@ -67,40 +64,48 @@ partial interface CanvasRenderingContext2D {
+
+#### The "linear-srgb" color space
 * Support is optional.
-* User agents that select either of the the p3 or the rec2020 color gamuts in CSS media queries must support this color space.
-* The color space corresponds to the DCI P3 specification (TODO: add link).
-* toDataURL/toBlob convert image data to the p3 space.
-* Images with no color profile, when drawn to the canvas, are assumed to be in the sRGB color space, and are converted to p3 for the purpose of the draw.
+* Requires that the pixelFormat attribute be "float16".
+* This color space exists to enable wide color gamut and high dynamic range applications that use physically correct linear color operations (e.g, blending and interpolation)
+* Color space definition
+   * This color space uses the same primaries as the sRGB color space.
+   * This color space has a linear transfer function that supports arbitrary pixel values, in particular, values outside of the range [0, 1].
+* Note that gradients will be physically uniform, and thus will not match gradients in an "srgb" canvas
+* Note that pixel values written by WebGL shaders are now in a space with a linear transfer function, and will not necessarily match those of an "srgb" canvas.
 
 #### The pixelFormat context creation attribute
 The pixelFormat attributes specifies the numeric types to be used for storing pixel values.
 * Support for "8-8-8-8" is mandatory. All other formats ar optional.
 * When an unsupported format is requested, the format shall fall back to "8-8-8-8".
-* With formats that use integer numeric types for the color channels, the canvas pixel buffer shall store gamma-corrected color values, using the transfer curves prescribed by the specification of the current color space.
-* With formats that use floating-point numeric types for the color channels, the canvas pixel buffer shall store non-gamma-corrected (a.k.a linear) color values.
-* Float values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels. How values outside of [0,1] are displayed depends on the capabilites of the device and output display. Some implementations may simply clamp these values to [0,1]. If the device and display are capable, a pixel value of (2,2,2) should be twice as bright as (1,1,1).
+* Float values outside of [0,1] range can be used to represent colors outside of the chosen color gamut. This allows float pixel formats to represent all possible colors and brightness levels.
+* How values outside of [0,1] are displayed depends on the capabilites of the device and output display.
+   * Standard dynamic range devices should clamp these values to [0, 1].
+   * For high dynamic range devices, pixel value of (2,2,2) in the "linear-srgb" color space should be should be twice as bright as (1,1,1).
```

Comment:
What is the assumed brightness of (1,1,1) and does that need to be specified here? How does a developer ensure alignment between (1,1,1) and other brightness values, e.g. HLG media playback? @FredrikHubinette has some thoughts on this.

Reply:
That's something that should be addressed consistently across CSS colors and media as well (and canvas should fall in line).